lib/daru/vector.rb

```diff
@@ -603,6 +603,31 @@ def sort opts={}, &block
       Daru::Vector.new(vector, index: index, name: @name, dtype: @dtype)
     end
 
+    # Sorts the vector according to it's`Index` values. Defaults to ascending
+    # order sorting.
+    #
+    # == Options
+    #
+    # ascending::
+    #   This can be `true` or `false`. False, if false, will sort `index` in
+    #   descending order. Defaults to true.
+    #
+    # == Returns:
+    #   new sorted `Vector` according to the index values.
+    #
+    # == Usage
+    #
+    #   dv = Daru::Vector.new [11, 12, 13], index: [23, 22, 21]
+    #   # Say you want to sort index in descending order
+    #   di.sort_by_index(ascending: false)
+    def sort_by_index opts={}
+      opts = {ascending: true}.merge(opts)
+      new_arrangement = resort_index(@index.each_with_index, opts)
```

Comment:
I am using resort_index similar to [sort](https://github.com/SciRuby/daru/pull/315/files#diff-8129d70abcb202c11af77e9269873527R598)